### DISCUSSION TITLE
Not work on Samsung Note 9 Android 9

### DISCUSSION THREAD

**Discussion by blitzermetal** [Mygod/VPNHotspot#590] [Q&A]
Hi guys,
I got Note 9 on Android 9 rooted with Magisk 17.
Install VPN hotspot but can get it work:
+ if I turn on the temporary hotspot, other devices can connect to hotspot and get the IP address but cannot connect to internet (while the host device still connect to internet)
+ if I turn on toggle wifi hotspot, it shows the swlan0 above, and the hotspot wifi is available, but the toggle of "wifi hotspot" and "swlan0" is still in disabled state. Other devices can connect to that hotspot wifi and able to connect to internet. But the traffic is not through the host's VPN.
already trying to disable "hardware tethering acceleration" and toggle the IP DHCP option but nothing works. 
Really need some helps, pls.

**Comment by Mygod** [comment]
Export debug information while everything is active.

**Reply by blitzermetal** [reply]
I manage to upgrade to android 10 and it works ok now. But there's still a problem:
I use 2 devices to connect to hotspot. Android device works perfectly but the iPhone always got IPv6 even I turn on the disable IPv6 leak option. How can i got it work on iphone?

**Reply by Mygod** [reply]
If you have the option turned on, the IPv6 traffic should be blocked even if they got the address.

**Reply by blitzermetal** [reply]
With the option disable IPv6 leak on, go to the website ifconfig.io, it show that the ip of my iphone is IPv6 and not through VPN. 
Is that safe?

**Reply by Mygod** [reply]
That sounds bad. Export debug information after reproducing this?

**Reply by blitzermetal** [reply]
Gonna upload the debug file when I'm able to reproduce it. Last time checking, everything seems fine for now
